File: app/src/main/java/com/absinthe/anywhere_/viewmodel/InitializeViewModel.java
package com.absinthe.anywhere_.viewmodel;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class InitializeViewModel extends ViewModel {

    public static final int ROOT_PERM = 1;
    public static final int SHIZUKU_CHECK_PERM = 2;
    public static final int SHIZUKU_PERM = 4;
    public static final int OVERLAY_PERM = 8;
    public static final int SHIZUKU_GROUP_PERM = 6;
    private MutableLiveData<Boolean> isRoot;
    private MutableLiveData<Boolean> isOverlay;
    private MutableLiveData<Boolean> isPopup;
    private MutableLiveData<Boolean> isShizukuCheck;
    private MutableLiveData<Boolean> isShizuku;
    private MutableLiveData<Integer> allPerm;

    public MutableLiveData<Boolean> getIsRoot() {
        if (isRoot == null) {
            isRoot = new MutableLiveData<>();
        }
        return isRoot;
    }

    public MutableLiveData<Boolean> getIsOverlay() {
        if (isOverlay == null) {
            isOverlay = new MutableLiveData<>();
        }
        return isOverlay;
    }

    public MutableLiveData<Boolean> getIsPopup() {
        if (isPopup == null) {
            isPopup = new MutableLiveData<>();
        }
        return isPopup;
    }

    public MutableLiveData<Boolean> getIsShizukuCheck() {
        if (isShizukuCheck == null) {
            isShizukuCheck = new MutableLiveData<>();
        }
        return isShizukuCheck;
    }

    public MutableLiveData<Boolean> getIsShizuku() {
        if (isShizuku == null) {
            isShizuku = new MutableLiveData<>();
        }
        return isShizuku;
    }

    public MutableLiveData<Integer> getAllPerm() {
        if (allPerm == null) {
            allPerm = new MutableLiveData<>();
            allPerm.setValue(0);
        }
        return allPerm;
    }

}

File: app/src/main/java/com/absinthe/anywhere_/viewmodel/InitializeViewModel.kt
package com.absinthe.anywhere_.viewmodel

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class InitializeViewModel : ViewModel() {

    var isRoot: MutableLiveData<Boolean> = MutableLiveData()
    var isOverlay: MutableLiveData<Boolean> = MutableLiveData()
    var isPopup: MutableLiveData<Boolean> = MutableLiveData()
    var isShizukuCheck: MutableLiveData<Boolean> = MutableLiveData()
    var isShizuku: MutableLiveData<Boolean> = MutableLiveData()
    var allPerm: MutableLiveData<Int> = MutableLiveData()

    companion object {
        const val ROOT_PERM = 1
        const val SHIZUKU_CHECK_PERM = 2
        const val SHIZUKU_PERM = 4
        const val OVERLAY_PERM = 8
        const val SHIZUKU_GROUP_PERM = 6
    }
}